### DISCUSSION TITLE
Disabling DeprecationWarning stderr logs

### DISCUSSION THREAD

**Discussion by lavsurgut** [apache/airflow#28924] [Q&A]
Hello guys. We are using Airflow 2.4.3 with KubernetesExecutor. First of all, thanks for supporting it and making it an amazing product. We have quite some legacy dags and when we run it in a new version in k8s, worker pods generate excessive amount of "DeprecationWarning" logs in stderr for every task, like this:

`DeprecationWarning: This operator is deprecated. Please use `BigQueryInsertJobOperator`.`

We tried to use just simple LOG_LEVEL setting for workers - but it disables all logs, including those that we are really interested in. It also does not format the log message, e.g. no name of a module or timestamp (looks like is coming from core python, e.g. Warning).
We tried to experiment with the custom logging config - but don't know yet what keys in config dictionary to change to disable just DeprecationWarning messages.

Also tried to pass `ignore::DeprecationWarning` flag to airflow entrypoint script. 

Do you know by chance how to disable it for all DAGs?

**Comment by Taragolis** [comment]
Did you tried to set env var [PYTHONWARNINGS](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONWARNINGS) to `ignore::DeprecationWarning:airflow.providers.google.cloud.operators.bigquery`?

I wouldn't recommend ignore this kind of warnings better to change current DAGs to use specific non-deprecated Operator.

**Reply by josh-fell** [reply]
+1 on updating your provider version and DAGs. `BigQueryInsertJobOperator` was [introduced 2+ years ago](https://github.com/apache/airflow/pull/8868).

**Reply by lavsurgut** [reply]
Yes, thats clear that its better to use proper operators. The only thing is that it's going to take a bit of time to do it. On the other side, I find quite disturbing that I can not switch off these warnings easily. 
setting env var [PYTHONWARNINGS](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONWARNINGS) to `ignore::DeprecationWarning:airflow.providers.google.cloud.operators.bigquery` did not help. 

I think i will try to understand where it is coming from on a worker side. I see that every task start is parsing/going through all dags every time, also not clear yet whats the reason its doing that.

**Reply by potiuk** [reply]
If it was easy to disable deprecation  warnings, which have clear (even if time consuming) path i sgoing totally against the whole idea of deprecation warnings. The whole idea is that they are annoying and it is not as easy as flipping a bool value to disable them. Of course this is not the case if the warnings are false positive - i.e. there is a valid case where you should still use (and continue to use) the old ways.  In this case I believe the pathe is clear and you'd imply should not use the old versions..

The whole idea is that you will have to invest time and effort to convert -and the question is how much annoying it is for you and how it accumulates so that finally you decide to invest and make the change. 

By easy disabling it not only you loose completely the incentive to update but also you silently avcumulate problems that you will have to fix later anyway when we delete those operators and you would like to use new operator from the same provider while you will not be able to use the old way. It is far better to deal with one such deprecation at a time than to defer it to the time when you will have no choice but fix 10 such problems.

Disabling such warning is actually - in this context - actively harming your organisation's long term interests IMHO so if you have no time for it, you should ask or budget of doing it because the more you delay it, the more harmful it will be for the organisation. Whether the time for that is good now and whether you accumulated enough of deprecations to spend budget now - is of course up to you, but disabling the warnings make you blind to actual extend of deprecations you have in the queue. You might the be surprised in a few months how big and harmful your update process will be.

Just saying. Surely you can have other motivations and problems but silencing deprecation warnings is almost always bad idea.

**Reply by lavsurgut** [reply]
Ok thanks guys for you input. Will accept it as an answer.
